### ISSUE TITLE
Add volume to compose.dev.yml so node_modules are found

### ISSUE THREAD

**Issue by nathanweeks** [legumeinfo/gcv#397]
For the development Compose file, a volume is also needed so /gcv/node_modules gets [populated with the contents of the container image](https://docs.docker.com/storage/volumes/#populate-a-volume-using-a-container) at that mount point:

```
--- a/compose.dev.yml
+++ b/compose.dev.yml
@@ -9,3 +9,4 @@ services:
       - .:/gcv
       - /gcv/dep
       - /gcv/scripts
+      - /gcv/node_modules
```
Otherwise, since the host directory gets bind-mounted to /gcv, there will be no node_modules directory when the `gcv` service starts:
```
$ docker compose -f compose.dev.yml up -d
...
$ docker compose -f compose.dev.yml ps
NAME                COMMAND                  SERVICE             STATUS              PORTS
gcv-gcv-1           "npx ng serve --host"   gcv                 exited (1)
$ docker compose -f compose.dev.yml logs
gcv-gcv-1  | npm ERR! could not determine executable to run
gcv-gcv-1  |
gcv-gcv-1  | npm ERR! A complete log of this run can be found in:
gcv-gcv-1  | npm ERR!     /root/.npm/_logs/2022-01-08T16_52_58_142Z-debug-0.log
```

**Comment by alancleary** [comment]
I can't recreate the error. Can you give more details, such as what's in the log? I'll note that using the command `docker compose -f compose.dev.yml logs` was showing outdated errors from an orphan container on my machine (confirmed with `logs -t`). Running the container without the daemon flag (`-d`) removed the outdated errors from the logs. (I should really just remove the orphan)

Regarding mounting the `node_modules` directory, it was previously mounted but I removed it from the compose file because it was causing builds to fail. I can't remember exactly why but I think it had something to do with the host and container having conflicting versions of node.

**Comment by nathanweeks** [comment]
The issue is that bind-mounting the root of the git working tree at /gcv in the `gcv` service container:
```
volumes:
   - .:/gcv
   - /gcv/dep
   - /gcv/dep
```
hides /gcv in the container. Subsequently, anonymous volumes for /gcv/dep and /gcv/dep make the contents of those directories as they exist in the container image available in the container, but /gcv/node_modules in the container image not exposed.

To reproduce, do a fresh `git clone` of the repo (or `git clean -xfd .` within your working tree, assuming you're on the "main" branch, to remove any node_modules directory that may exist in the working tree as the result of local development). In the root of the git working tree, execute `docker compose -f compose.dev.yml up -d`. The "gcv" service exits with a non-zero exit status as listed above, and the `docker compose -f compose.dev.yml. logs` command shows the above command.

**Comment by alancleary** [comment]
OK, thanks for clarifying. I was able to reproduce the issue with your instructions. Interestingly, I can't seem to reproduce the issue I was having before when `node_modules` was listed as a volume in the compose file...

Here's a question that will certainly reveal my feeble understanding of what's actually happening here: if we add `/gcv/node_modules` as a volume in the compose file then should `node_modules` be removed from the `.dockerignore` file? I added it to ensure docker makes a clean build every time...

**Comment by nathanweeks** [comment]
> Here's a question that will certainly reveal my feeble understanding of what's actually happening here: if we add `/gcv/node_modules` as a volume in the compose file then should `node_modules` be removed from the `.dockerignore` file? I added it to ensure docker makes a clean build every time...

A robust question :-)  [.dockerignore](https://docs.docker.com/engine/reference/builder/#dockerignore-file) affects only the build context during the container image build process (`docker build` or `docker compose build` commands), and not any run-time container instance host bind mounts (e.g., specified manually specified by `docker -v`, or as a `volume` or in a Compose file).

**Comment by alancleary** [comment]
Fixed in commit dc24267bca2ef213c95413a1d0bd802aa5d1c104.

@ATintern you should be able to run the client development server now:
```
docker-compose -f compose.dev.yml up -d
```
If it's still not working, you may need to force docker compose to rebuild:
```
docker-compose -f compose.dev.yml up -d --build --force-recreate
```
